samtranslator/model/eventsources/push.py

```diff
@@ -122,8 +123,18 @@ def to_cloudformation(self, **kwargs):
         resources.append(events_rule)
 
         events_rule.ScheduleExpression = self.Schedule
+
+        if self.State and self.Enabled:
+            raise InvalidEventException(self.relative_id, "State and Enabled Properties cannot both be specified.")
+
+        if self.State:
+            events_rule.State = self.State
+
         if self.Enabled is not None:
-            events_rule.State = "ENABLED" if self.Enabled else "DISABLED"
+            if isinstance(self.Enabled, bool):
```

Comment:
If we are introducing a new property to get the behavior right with intrinsic support, why don't we just leave this as it was before?

Reply:
I meant to revert this back but looks like it got lost in me putting this task up and down. Will update.